### PR TITLE
Fixed odo app list json output

### LINKED ISSUE #1327: odo app list list -o json does not output json
[kind/bug]

## What versions of software are you using?
- Operating System: Win10 Pro
- Output of `odo version`:

```
Server: https://172.17.247.235:8443
OpenShift: v3.9.31
Kubernetes: v1.9.1+a0ce1bc657
```

## How did you run odo exactly?
`odo app list -o json`

## Actual behavior

```
$ odo app list -o json
There are no applications deployed in the project 'myproject'.
```

## Expected behavior

```
$ odo app list -o json
Some JSON output
```

### PR DESCRIPTION
**Pull request by surajnarwade**
## What is the purpose of this change? What does it change?

odo will return empty struct if there's no app

## Was the change discussed in an issue?
fixes #1327

## How to test changes?

validate the output for below commands
`odo app list -o json`

### CODE REVIEW

**Comment by jeffmaury** [comment]
Works for me

**Comment by surajnarwade** [comment]
@amitkrout , Added test

**Comment by amitkrout** [comment]
@surajnarwade UTs are missing for freshly created function ```GetMachineReadableFormat``` and ```GetMachineReadableFormatForList```

Integration test looks good to me except this missing UT coverage.

File: pkg/application/application.go [hunk 1]
```diff
@@ -258,3 +260,41 @@ func Exists(client *occlient.Client, appName string) (bool, error) {
...
+func GetMachineReadableFormat(client *occlient.Client, appName string, projectName string, active bool) App {
+	componentList, _ := component.List(client, appName)
+	var compList []string
+	for _, comp := range componentList.Items {
+		compList = append(compList, comp.Name)
+	}
+	appDef := App{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "app",
+			APIVersion: "odo.openshift.io/v1alpha1",
```
Comment (cdrage) [inline pkg/application/application.go]: these values, shouldn't they be constants?

File: pkg/application/application.go [hunk 2]
```diff
@@ -258,3 +260,41 @@ func Exists(client *occlient.Client, appName string) (bool, error) {
...
+	}
+	return appDef
+}
+
+// GetMachineReadableFormatForList returns application list in machine readable format
+func GetMachineReadableFormatForList(apps []App) AppList {
+	return AppList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "List",
+			APIVersion: "odo.openshift.io/v1alpha1",
```
Comment (cdrage) [inline pkg/application/application.go]: Should this be a constant (as it may purposely graduate it from alpha one day...)

**Comment by surajnarwade** [comment]
merge after CI is :green_heart:

**Pull request merged**